pkg/kubelet/pod_workers.go

```diff
@@ -882,6 +922,8 @@ func (p *podWorkers) completeTerminatingRuntimePod(pod *v1.Pod) {
 	delete(p.podUpdates, pod.UID)
 	delete(p.lastUndeliveredWorkUpdate, pod.UID)
 	delete(p.terminatingStaticPodFullnames, kubecontainer.GetPodFullName(pod))
+
+	p.startNextStaticPod(kubecontainer.GetPodFullName(pod))
```

Comment:
Why is this needed now?  isAllowedToStart should be able to determine this by checking whether it's the first allowed pod in the list of pending.